credoai/modules/dataset_modules/dataset_base.py

```diff
@@ -61,4 +63,44 @@ def _make_pipe(self):
                [('scaler', StandardScaler()),
                 ('model', model)])
         return pipe
+
+    def _assess_proxy_in_feature_space(
+        self, excluded_columns=None, method="mutual_information"
+    ):
+        """Assesses features based on how much they serve as a proxy for a sensitive attribute
+        This method relies on features information only (label information is not used)
+
+        Parameters
+        ----------
+        excluded_columns : list[str]
+            List of columns to exclude from the assessment
+        method : str, optional
+            Method to use for quantifying the magnitude of proxy, by default 'mutual_information'
+            Choices:
+                'mutual_information' (the "amount of information" in normalized nats obtained about a feature by observing another feature)
+
+        Returns
+        -------
+        dict
+            Keys are feature names and values are their proxy scores
+            Score between 0.0 and 1.0
+        """
+        proxy_scores = {}
+        
+        if method == "mutual_information":
+            # Exclude features of type float
+            float_cols = list(self.X.select_dtypes(include=["float16", "float32", "float64"]))
+            if excluded_columns:
+                excluded_columns.extend(float_cols)
+            else:
+                excluded_columns = float_cols
+
+            for column in self.X.drop(excluded_columns, axis=1):
+                proxy_scores[column] = normalized_mutual_info_score(
+                    self.X[column], self.sensitive_features
+                )
+
+        return proxy_scores
```

Comment:
Once they have these scores, how do they know if something is a proxy or not? Is there a cutoff?

Reply:
Being a proxy is not a binary state.

Reply:
I agree, but I am thinking of a user who has to decide whether to remove or keep a variable.